src/sources/splunk_hec.rs

```diff
@@ -1272,6 +1263,27 @@ mod tests {
         );
     }
 
+    #[tokio::test]
+    async fn whitespace() {
+        trace_init();
+
+        let message = r#" {"event":"first"} "#;
+        let (source, address) = source().await;
+
+        assert_eq!(
+            200,
+            post(address, "services/collector/event", message).await
+        );
+
+        let event = collect_n(source, 1).await.remove(0);
+        assert_eq!(event.as_log()[log_schema().message_key()], "first".into());
+        assert!(event.as_log().get(log_schema().timestamp_key()).is_some());
+        assert_eq!(
+            event.as_log()[log_schema().source_type_key()],
+            "splunk_hec".into()
+        );
+    }
```

Comment:
I suggest to add a test for trailing newlines `\n` as well as trailing whitespaces.
This is the case of the Splunk Kafka Connect plugin, which sends every JSON object with a trailing newline.
The source should be resilient to both, trailing whitespaces and trailing newlines.

Reply:
Good call, I missed the mention of newlines in the issue. I pushed 60b4baac6 which adds newlines to the test input here.

Reply:
There is actually still whitespace here

```rust
let message = r#"
        {"event":"first"}
        "#;
```

The leading whitespace on the next line will be included so the input will end up being:

```

        {"event":"first"}

```

Reply:
Oh I see. Apologies as non-printable characters are hard to see in GH for me :)
I am still not sure how the new messsage tests _trailing_ whitespace as it did before:
```rust
let message = r#"{"event":"first"} "#;
```
The original issue did not have a problem with whitespace/newline in the middle so I think the safer/stronger test would be to have each case separatedly tested.

Reply:
Very fair 😄 I did neglect I dropped the trailing whitespace case when adding the newline case. I pushed 92aba6944 which tests both separately.

Reply:
Looks easier to follow now! Thanks for the patience and sorry for the annoyance :D
And many thanks for implementing the fix 💯